applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp

```diff
@@ -409,73 +437,97 @@ namespace MPMParticleGeneratorUtility
 
-                                p_condition->SetValuesOnIntegrationPoints(MPC_DISPLACEMENT, { mpc_displacement }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_DISPLACEMENT, { mpc_imposed_displacement }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_VELOCITY, { mpc_velocity }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_VELOCITY, { mpc_imposed_velocity }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_ACCELERATION, { mpc_acceleration }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_ACCELERATION, { mpc_imposed_acceleration }, process_info);
-                                // Mark as boundary condition
-                                p_condition->Set(BOUNDARY, true);
+                                    // Create new material point condition
+                                    new_condition_id = last_condition_id + 1 ;
+                                    
+                                    Condition::Pointer p_condition = new_condition.Create(
+                                        new_condition_id, p_quadrature_point_geometry, properties);
+                                
+                                    ParticleConditions.push_back(p_condition);
+                                    
+                                    ProcessInfo process_info = ProcessInfo();
+
+                                    // Setting particle condition's initial condition
+                                    //p_condition->SetValuesOnIntegrationPoints(MPC_CONDITION_ID, mpc_condition_id, process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_COORD, {xg} , process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_AREA,  mpc_area  , process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_NORMAL, { mpc_normal }, process_info);
+
+
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_DISPLACEMENT, { mpc_displacement }, process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_DISPLACEMENT, { mpc_imposed_displacement }, process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_VELOCITY, { mpc_velocity }, process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_VELOCITY, { mpc_imposed_velocity }, process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_ACCELERATION, { mpc_acceleration }, process_info);
+                                    p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_ACCELERATION, { mpc_imposed_acceleration }, process_info);
```

Comment:
`MPC_DISPLACEMENT`, `MPC_VELOCITY` and `MPC_ACCELERATION` are zero... can we remove the corresponding lines or the initialization to zero is necessary?

Reply:
I think we should keep this initialization - its similar in the elements as well

Reply:
Ok, but there is something that is not clear to me (even if you are not changing this part with this PR)

Here we are assigning `mpc_contact_force` to `POINT_LOAD` when `is_interface == true` and then few lines below a new value is assigned to `POINT_LOAD`. I think that in the first case the variable should be `MPC_CONTACT_FORCE`, right?

Moreover, `MPC_NORMAL` is not assigned in this case (`MPMParticlePointLoadCondition`). Why?

https://github.com/KratosMultiphysics/Kratos/blob/1bda0ce046a36c3c24d29a4d8a3830e26e29f785/applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp#L410-L424

Reply:
Thanks for that hint - I removed the line where mpc_contact_force is assigned to the POINT_LOAD